src/test/node/images/controllers.js

```diff
@@ -19,46 +19,36 @@ const MockImageB = {
   caption: 'Hello World'
 };
 test('get all image of an item', t => {
-  const itemKey = 'item-BlaBla';
-  const imageIndexKey = `image-${STATE.ALIVE}-${itemKey}-${MockImageA.key}`;
-
-  clearDB().then(() => {
-    const opts = [];
-    opts.push({
-      type: 'put',
-      key: imageIndexKey,
-      value: {
-        key: MockImageA.key
+  const itemKey = mockItems[0].key;
+  clearDB().then(initMock).then(() => {
+    const req = httpMocks.createRequest({
+      method: 'GET',
+      url: '/api/images',
+      query: {
+        item: itemKey
       }
     });
-    opts.push({
-      type: 'put',
-      key: MockImageA.key,
-      value: MockImageA
-    });
+    const res = httpMocks.createResponse();
 
-    testDB.batch(opts, (err) => {
-      if (err) {
-        t.fail('database batch error');
-        t.end();
-        return;
+    controller.getAll(req, res, () => {
+      const values = res._getData().values;
+      if (values.length === 0) {
+        t.fail('empty values array');
       }
-      const req = httpMocks.createRequest({
-        method: 'GET',
-        url: '/api/images',
-        query: {
-          item: itemKey
-        }
-      });
-      const res = httpMocks.createResponse();
-
-      controller.getAll(req, res, () => {
-        const value = res._getData().values;
-        t.equal(value[0].caption, MockImageA.caption, 'should be same caption');
-        t.equal(value[0].key, MockImageA.key, 'should be same key');
-        t.equal(typeof (value[0].url), 'string', 'should be same type');
-        t.end();
+      values.map((value) => {
+        /* eslint-disable curly */
```

Comment:
You can remove if statement.
tape.ok pass when the value is not a [0, null, undefined]
```
        t.equal(value.itemKey, itemKey, 'item key should be equal');
        t.ok(value.caption, 'no caption field');
        t.ok(value.key, 'no key field');
        t.ok(value.user, 'no user field');
        t.ok(value.userKey, 'no userKey field');
        t.ok(value.createdDate, 'no createdDate field');
        t.ok(value.user.key, 'no user.key field');
        t.ok(value.user.name, 'no user.name field');
        t.ok(value.user.email, 'no user.email field');
```